eventorder_test.go

```diff
@@ -98,48 +103,46 @@ func TestEventOrderGuarantee(t *testing.T) {
 		"workspaceId": workspaceID,
 	}
 	configJsonPath := workspaceConfig.CreateTempFile(t, "testdata/eventOrderTestTemplate.json", templateCtx)
-	t.Setenv("RSERVER_BACKEND_CONFIG_CONFIG_FROM_FILE", "true")
-	t.Setenv("RSERVER_BACKEND_CONFIG_CONFIG_JSONPATH", configJsonPath)
+	config.Set("BackendConfig.configFromFile", true)
```

Comment:
Starting a service as a separate process comes with disadvantages too. Please see [this comment](https://github.com/rudderlabs/rudder-server/pull/2439#discussion_r978286582) and let me know if you wish me to elaborate further

Reply:
@atzoum it sure has disadvantages like harder to debug (setting breakpoints) and code verbosity (setting up the exec, piping output, etc...). have you come across more issues? I'd like to know because in some places it helped me with isolation.

Reply:
Practically its only advantage is isolation, everything else could be considered as a disadvantage, e.g. a non-restrictive list:
1. debugging (practically impossible)
2. extensibility (unit testing components, i.e. non-programms)
3. intuitiveness (a test case executing `go run` causes some people to say WTF)
4. configuration (expecting that the version of the go binary executing the test is the same as the go binary available in the path)

Reply:
Good point about the configuration. About extensibility one thing is to do an integration test with it, one thing a unit test. The latter definitely not recommended :slightly_smiling_face: